eventdata/parameter_sources/randomevent.py

```diff
@@ -234,12 +235,32 @@ def add_fields(self, event):
         event['httpversion'] = data[5]
 
 
+def convert_to_bytes(size):
+    matched_size = re.match(r"^(\d+)\s?(kB|MB|GB)?$", size)
```

Comment:
A possible speed improvement (since this is in the hot code path) for a future iteration would be to replace the regexp with a conditional that:

1. first checks `size.isdigit()` and if true returns value as is
2. if 1 false, sets `value=size[:-2]` and `unit=size[-2:]` and proceeds with the unit checks.

Reply:
I don't think this is on the hot code path? This function should only be used in the constructor of the timestamp generator which is created upon setup of the parameter source?

Reply:
ah great point. Let's ignore this optimization suggestion then.